A72-open-telemetry-tracing.md

```diff
@@ -0,0 +1,659 @@
+   * 
+   *     
+   * Add interceptors and StreamTracerFactory obtained from the module to GlobalInterceptors.    
+   * 
+   *    GlobalInterceptors.setInterceptors(
+   *      Arrays.asList(otModule.getClientTracingInterceptor()),
+   *      Arrays.asList(otModule.getServerTracerFactory()));        
+   */
+  
+  // Return the client interceptor to be installed by users to generate trace information.
+  public ClientInterceptor getClientTracingInterceptor();
+
+  // Return the ServerStreamTracerFactory to be installed by users to generate trace information.
+  public ServerStreamTracerFactory getServerTracerFactory();
+}
+
+```
+
+### C++
+The following new methods will be added in `OpenTelemetryPluginBuilder`.
+
+```C++
+class OpenTelemetryPluginBuilder {
+ public:
+  // If `SetTracerProvider()` is not called, no traces are collected.
+  OpenTelemetryPluginBuilder& SetTracerProvider(
+      std::shared_ptr<opentelemetry::metrics::TracerProvider> tracer_provider);
+  // Set one or multiple text map propagators for span context propagation, e.g.
+  // the community standard ones like W3C, etc.
+  OpenTelemetryPluginBuilder& SetTextMapPropagator(
+      std::unique_ptr<opentelemetry::context::propagation::TextMapPropagator>
+          text_map_propagator);
+};
+```
+
+### Go
+The following methods will be added to the opentelemetry package.
+```go
+import (
+  "go.opentelemetry.io/otel/trace"
+)
+
+// TraceOptions are the trace options for OpenTelemetry instrumentation.
+type TraceOptions struct {
+  TraceProvider trace.TraceProvider
+}
+
+// DialOption returns a dial option which enables OpenTelemetry instrumentation
+// code for a grpc.ClientConn.
+//
+// Client applications interested in instrumenting their grpc.ClientConn should
+// pass the dial option returned from this function as a dial option to
+// grpc.Dial().
+//
+// Using this option will always lead to instrumentation, however in order to
+// use the data a SpanExporter must be registered with the TraceProvider in the
+// TraceOption. Client side has retries, so a Unary and Streaming Interceptor are
+// registered to handle per RPC traces, and a Stats Handler is registered to handle
+// per RPC attempt trace. These three components registered work together in
+// conjunction, and do not work standalone.
+func DialOption(to TraceOptions) grpc.DialOption {}
+
+// ServerOption returns a server option which enables OpenTelemetry
+// instrumentation code for a grpc.Server.
+//
+// Server applications interested in instrumenting their grpc.Server should pass
+// the server option returned from this function as an argument to
+// grpc.NewServer().
+//
+// Using this option will always lead to instrumentation, however in order to
+// use the data a SpanExporter must be registered with the TraceProvider option.
+// Server side does not have retries, so a registered Stats Handler is the only
+// option that is returned.
+func ServerOption(to TraceOptions) grpc.ServerOption {}
+```
+
```

Comment:
@XuanWang-Amos I think it would be appropriate to add a `###Python` section here about API. The propagator part it would be similar to java/go, please review.